frontend/js/src/search/SongSearch.tsx

```diff
@@ -0,0 +1,166 @@
+import { useQuery } from "@tanstack/react-query";
+import * as React from "react";
+import _ from "lodash";
+import { useSearchParams } from "react-router-dom";
+import GlobalAppContext from "../utils/GlobalAppContext";
+import Loader from "../components/Loader";
+import ListenCard from "../common/listens/ListenCard";
+import { getObjectForURLSearchParams } from "../utils/utils";
+import Pagination from "../common/Pagination";
+
+const RECORDING_COUNT_PER_PAGE = 50;
+
+type SongSearchProps = {
+  searchQuery: string;
+};
+
+type SongTypeSearchResult = {
+  count: number;
+  offset: number;
+  recordings: {
```

Comment:
You might be able to use the existing MusicBrainzRecording, MusicBrainzRelease, etc. types that exist

Reply:
I did checkout the types which exist right now, but I don't think we can re-use any of those.